presto-orc/src/main/java/com/facebook/presto/orc/metadata/Footer.java

```diff
@@ -27,14 +32,20 @@
     private final List<StripeInformation> stripes;
     private final List<OrcType> types;
     private final List<ColumnStatistics> fileStats;
+    private final Map<String, Slice> userMetadata;
 
-    public Footer(long numberOfRows, int rowsInRowGroup, List<StripeInformation> stripes, List<OrcType> types, List<ColumnStatistics> fileStats)
+    public Footer(long numberOfRows, int rowsInRowGroup, List<StripeInformation> stripes, List<OrcType> types, List<ColumnStatistics> fileStats, Map<String, Slice> userMetadata)
     {
         this.numberOfRows = numberOfRows;
         this.rowsInRowGroup = rowsInRowGroup;
         this.stripes = ImmutableList.copyOf(requireNonNull(stripes, "stripes is null"));
         this.types = ImmutableList.copyOf(requireNonNull(types, "types is null"));
         this.fileStats = ImmutableList.copyOf(requireNonNull(fileStats, "columnStatistics is null"));
+        requireNonNull(userMetadata, "userMetadata is null");
+        this.userMetadata = ImmutableMap.copyOf(userMetadata.entrySet().stream().collect(toMap(
```

Comment:
nice - I opted for streams API instead for guava API. But this is more readable for sure :)